docs/porting/libraries.md

```diff
@@ -1,5 +1,151 @@
+
+Here's the essentials you need to know about `project.json`:
+
+1. You need to restore packages to generate a `project.lock.json` the first time you create the project.
+
+   In Visual Studio, you can right-click the project and select "Restore Packages".
+   
+   If you're using the .NET CLI, you'll need to navigate to the project folder and type:
+   
+   `$ dotnet restore`
+
+2. Any further changes to the `project.json` will require restoring packages to generate a new `project.lock.json` file.
+3. The `project.lock.json` file is what the compiler reads when you build your project.
+
+To learn more about `project.json`, please refer to [the documentation] on it.
+
+Once you can build your .NET Core project(s) successfully, you can begin porting code over!
+
+## Actually porting your code
+
+Finally, porting the code itself!  Depending on how your .NET Framework code is structured, this could actually be the easiest part of porting over to .NET Core.  How you actually go about doing this will depend highly on the way your code is structured.
+
+### Getting a testing framework for testing as you port
+
+The best way to make sure everything works when you've ported your code is to *test your code as you port it to .NET Core*.  To do this, you'll need to use a testing framework that will build and run tests for .NET Core.  Currently, you have two options:
+
+* Xunit
+   - TODO: Instructions for Visual Studio
+   - TODO: Instructions for .NET CLI
+* Nunit
+  - TODO: Instructions for Visual Studio
+  - TODO: Instructions for .NET CLI
+
+If you've been using Xunit or Nunit to write your tests for your library, great!  This shouldn't be much work.  However, if you've been using the built-in testing framework that ships with Visual Studio, you'll need to migrate your existing tests or write new tests in either Xunit or Nunit.
+
+### Recommended Approach
+
+A good way to port your code is to begin with the "base" of your library.  This may be data models or some other foundational classes and methods that everything else uses directly or indirectly.
+
+1. Copy over the "base" of your library into the new .NET Core project.
+   a. Rename the namespaces if you wish to have both .NET Core and .NET Framework projects.
+2. Make any changes needed to get the code to compile.
+   a. Much of this may just require adding NuGet package dependencies to your `project.json`.
+   b. It may helpful to comment out old code as you slowly get things compiling.
+3. Port over any old tests which consume what you've just ported.
+   a. If you're using an aditional testing library that doesn't run on .NET Core, you may have to rewrite some tests or go forward without tests temporarily.
+   b. Run or rewrite tests as needed.
+4. Pick the next layer of code to port over and repeat steps 2 and 3!
+
+If you methodically move "up" from the "base" of your library, testing each layer as needed, you shouldn't run into unexpected issues.
+
```

Comment:
I think the problem with this is that going back and forth between the report and VS takes *forever*.  I used the approach of ApiPort to analyze violations and develop a plan for how to port and it worked out pretty well.  If Moq was available for .NET Core, I also would have run tests as I was porting as well.  That part actually annoyed me when I ported over ... I had all these tests that mocked network calls but I couldn't run them!

I think we may need to have a chat about what we feel is a good overall approach.  Your approach seems to be centered around the tooling and fairly procedural in nature.  The approach I'm suggesting is using the tooling to form your own plan and working your way up from a compilation basis.  I think there's a happy medium to be explored here.